lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2372,8 +2372,25 @@ namespace {
     }
 
     Decl *VisitNamespaceDecl(const clang::NamespaceDecl *decl) {
-      // FIXME: Implement once Swift has namespaces.
-      return nullptr;
+      // If we have a name for this declaration, use it.
+      Optional<ImportedName> correctSwiftName;
+      auto importedName = importFullName(decl, correctSwiftName);
+      if (!importedName) return nullptr;
+
+      auto dc =
+          Impl.importDeclContextOf(decl, importedName.getEffectiveContext());
+      if (!dc)
+        return nullptr;
+
+      SourceLoc loc = Impl.importSourceLoc(decl->getBeginLoc());
+
+      // FIXME: If Swift gets namespaces, import as a namespace.
+      auto enumDecl = Impl.createDeclWithClangNode<EnumDecl>(
```

Comment:
What happens if you open the same namespace twice? Forget about separate modules now, even just the one-file version of this seems like a problem.

Reply:
This works. When importing the namespace decl, it gets canonicalized to be a single decl. If all you want to do is import it as a context type, it doesn't really need to do much. When doing name-lookup, these assumptions might become a problem, but they will definitely be a problem for multiple modules.

Reply:
> When importing the namespace decl, it gets canonicalized to be a single decl.

Ah, I forgot that `getCanonicalDecl` happens. Okay, makes sense.